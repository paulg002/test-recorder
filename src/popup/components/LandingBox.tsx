import * as React from 'react';
import CypressCone from './CypressCone';

export interface LandingBoxProps { isValidTab: boolean }

export default ({ isValidTab }: LandingBoxProps) => (
  <div id="landing-box">
    <CypressCone />
    {(
      isValidTab
        ? <p className="text">Click &apos;Start Recording&apos; to start recording!</p>
        : <p className="text">This is not a valid page for recording.</p>
    )}
  </div>
);
