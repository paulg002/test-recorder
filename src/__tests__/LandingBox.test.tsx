import * as React from 'react';
import { shallow } from 'enzyme';
import LandingBox from '../popup/components/LandingBox';
import '../setupTests';

describe('LandingBox', () => {
  let wrapper;

  it('Text should display "Click Start Recording to start" when isValidTab is true', () => {
    wrapper = shallow((<LandingBox isValidTab />));
    expect(wrapper).toMatchSnapshot();
    const textArea = wrapper.find('.text');
    expect(textArea.text()).toBe('Click \'Start Recording\' to start recording!');
  });
  it('Text should display "This is not a valid page for recording" when isValidTab is false', () => {
    wrapper = shallow(<LandingBox isValidTab={false} />);
    const textArea = wrapper.find('.text');
    expect(textArea.text()).toBe('This is not a valid page for recording.');
  });
});
