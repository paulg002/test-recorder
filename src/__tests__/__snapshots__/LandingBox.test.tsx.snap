// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingBox Text should display "Click Start Recording to start" when isValidTab is true 1`] = `
<div
  id="landing-box"
>
  <CypressCone />
  <p
    className="text"
  >
    Click 'Start Recording' to start recording!
  </p>
</div>
`;
